requests/utils.py

```diff
@@ -615,15 +615,15 @@ def set_environ(env_name, value):
     if value is not None:
         old_value = os.environ.get(env_name)
         os.environ[env_name] = value
-    try:
+        try:
+            yield
+        finally:
+            if old_value is None:
+                del os.environ[env_name]
+            else:
+                os.environ[env_name] = old_value
+    else:
         yield
```

Comment:
Hrm, I don't love the two separate `yield` statements here. Do you mind refactoring the old `finally` statement to invert the conditional on `if value is None: return` and to instead have `if value is not None` then do the dance with `old_value`.